http-custom-mechanism/configure-security-domain.cli

```diff
@@ -2,7 +2,11 @@
 
 # Start batching commands
 batch
-#  Map the application security domain to an Elytron HTTP authentication factory.
+
+# Configure an http authentication factory
+/subsystem=elytron/http-authentication-factory=application-http-authentication:add(http-server-mechanism-factory=global,security-domain=ApplicationDomain,mechanism-configurations=[{mechanism-name=BASIC,mechanism-realm-configurations=[{realm-name=Simple Realm}]},{mechanism-name=FORM}])
```

Comment:
@SoniaZaldana The default WildFly config used to contain an `application-http-authentication` HTTP authentication factory. A little while ago, we removed this from the default config and we added the ability to specify a `security-domain` directly from the `application-security-domain` resource in the Undertow subsystem instead of needing to reference an `http-authentication-factory`. For simple cases where a deployment can make use of the standard HTTP authentication mechanisms (i.e., BASIC, CLIENT_CERT, DIGEST, FORM), we can map the application security domain to an Elytron security domain as follows:

```
/subsystem=undertow/application-security-domain=other:add(security-domain=ApplicationDomain)
```

Since the first part of the quickstart demonstrates how to use BASIC authentication, we should use this simple command instead of adding the `http-authentication-factory` and referencing that from the `application-security-domain` mapping.

The second part of the quickstart then shows how to make use of a custom HTTP authentication mechanism. In this case, `configure-elytron.cli` shows how to add an appropriate `http-authentication-factory` and update the `application-security-domain` mapping to reference that. This script will need to be updated to undefine the `security-domain` attribute from the `application-security-domain` mapping prior to configuring the `http-authentication-factory` attribute.

Reply:
Thanks for the through explanation! I've updated the code.

Reply:
Thanks, Sonia! Looks good. Just one more minor change will be needed to update the following sentence in the "Configure the Application Security Domain" section in the README:

"You need to map this security domain to an Elytron HTTP authentication factory."

Reply:
Oh right, sorry I missed that. I updated the README.